### PR TITLE
contracts/swap, swap: refactor contractAddress, remove backend as inputParam, function order in contracts/swap

### PR DESCRIPTION
**Pull request by Eknir**
This PR includes a small refactor with regards to where the chequebook contractAddress is stored.

Previously, it was stored explicitly in `swap.owner.Contract`, while it was also implicitly stored in swap.contract (otherwise we would not be able to do calls to the contract).

This PR removes `swap.owner.Contract` as a place to store the contractAddress and exposes the contractAddress via `swap.getParams()`.

Additionally, two other refactors were included:

- remove backend as an input parameter in all `swap` functions which have `swap` as a receiver (instead, use swap.backend)
- change the order of functions in contracts/swap, to reflect the order of the interface definition

### CODE REVIEW

**Review by mortelli — APPROVED** [review]
LGTM. just one question

**Review by zelig — APPROVED** [review]

File: swap/protocol_test.go [hunk 1]
```diff
@@ -202,6 +202,11 @@ func TestTriggerPaymentThreshold(t *testing.T) {
 	debitorSwap, clean := newTestSwap(t, ownerKey)
 	defer clean()
 
+	ctx := context.Background()
+	err := testDeploy(ctx, debitorSwap)
+	if err != nil {
+		t.Fatal(err)
+	}
```
[the remark this answers is not in the document]
Comment (Eknir) [reply swap/protocol_test.go:209]: If we would not set this, we would at one point get a nil pointer dereference error. Specifically:
The Add fuction will trigger the `createCheque` function (`peer.go:121`). Here, we now have call `p.swap.GetParams()`, which will fail if `p.contract` is not initialized. Previously, we could get away with this as the value of an (uninitialized) `swap.owner.ContractAddress` would just be 0.

 You can easily test this by just removing these lines of code and run the test ;)

[the remark this answers is not in the document]
Comment (mortelli) [reply swap/protocol_test.go:209]: cleared up, thanks!

**Review by santicomp2014 — APPROVED** [review]
LGTM.

File: swap/swap.go [hunk 2]
```diff
@@ -130,7 +129,7 @@ func createOwner(prvkey *ecdsa.PrivateKey) *Owner {
 
 // DeploySuccess is for convenience log output
 func (s *Swap) DeploySuccess() string {
-	return fmt.Sprintf("contract: %s, owner: %s, deposit: %v, signer: %x", s.owner.Contract.Hex(), s.owner.address.Hex(), s.params.InitialDepositAmount, s.owner.publicKey)
+	return fmt.Sprintf("contract: %s, owner: %s, deposit: %v, signer: %x", s.GetParams().ContractAddress.Hex(), s.owner.address.Hex(), s.params.InitialDepositAmount, s.owner.publicKey)
```
Comment (holisticode) [inline swap/swap.go:132]: Not related to this PR, but your change reminds me of it: `signer: %x........,s.owner.publickey` actually prints garbage. Can you open an issue where we'd print the hex value of it? Thanks.

As it is a very small change, you could do it as part of this PR if you want.

**Review by holisticode — APPROVED** [review]
Very nice PR, it brings more simplification and clarity to the actual Swap elements.

**Pull request merged**